crates/binjs_io/src/json/value.rs

```diff
@@ -0,0 +1,216 @@
+/// JS value in JSON.
+use TokenReaderError;
+
+use binjs_shared::SharedString;
+
+use std::borrow::Borrow;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+/// JS value in JSON.
+pub enum Value {
```

Comment:
It's a bit annoying to have two different implementations of JSON in the code. Could you document in the file the difference between this implementation and the one in json.rs?

Reply:
Also, `Value` is read-only. I imagine that's by design, right?

Reply:
yeah, I'll add comment.